packages/instant/webpack.config.js

```diff
@@ -11,22 +12,107 @@ const GIT_SHA = childProcess
     .toString()
     .trim();
 
-const HEAP_PRODUCTION_ENV_VAR_NAME = 'INSTANT_HEAP_ANALYTICS_ID_PRODUCTION';
-const HEAP_DEVELOPMENT_ENV_VAR_NAME = 'INSTANT_HEAP_ANALYTICS_ID_DEVELOPMENT';
-const getHeapAnalyticsId = modeName => {
-    if (modeName === 'production') {
-        return process.env[HEAP_PRODUCTION_ENV_VAR_NAME];
+const getEnvironmentName = (env, argv) => {
+    if (env && env.dogfood) {
+        return 'dogfood';
+    } else if (env && env.staging) {
+        return 'staging';
     }
 
-    if (modeName === 'development') {
-        return process.env[HEAP_DEVELOPMENT_ENV_VAR_NAME];
+    // argv.mode should be 'development' or 'production'
+    return argv.mode;
+};
+
+const getHeapAnalyticsId = environmentName => {
+    if (environmentName === 'production') {
+        return process.env['INSTANT_HEAP_ANALYTICS_ID_PRODUCTION'];
+    }
+
+    if (environmentName === 'development' || environmentName === 'dogfood' || environmentName === 'staging') {
+        return process.env['INSTANT_HEAP_ANALYTICS_ID_DEVELOPMENT'];
     }
 
     return undefined;
 };
 
+const ROLLBAR_PUBLISH_TOKEN_ENV_NAME = 'INSTANT_ROLLBAR_PUBLISH_TOKEN';
+const ROLLBAR_CLIENT_TOKEN_ENV_NAME = 'INSTANT_ROLLBAR_CLIENT_TOKEN';
+const getRollbarPlugin = environmentName => {
+    if (!environmentName) {
+        return undefined;
+    }
+
+    const publishToken = process.env[ROLLBAR_PUBLISH_TOKEN_ENV_NAME];
+    if (!publishToken) {
+        return undefined;
+    }
+
+    let rollbarPublicPath;
+    if (environmentName === 'dogfood') {
+        rollbarPublicPath = 'http://0x-instant-dogfood.s3-website-us-east-1.amazonaws.com';
+    } else if (environmentName === 'staging') {
+        rollbarPublicPath = 'http://0x-instant-staging.s3-website-us-east-1.amazonaws.com';
+    } // TODO(sk): When we decide on JS cdn, add public path here
+
+    if (!rollbarPublicPath) {
+        console.log('No rollbar public path');
+        return undefined;
+    }
+
+    const rollbarPluginOptions = {
+        accessToken: publishToken,
+        version: GIT_SHA,
+        publicPath: rollbarPublicPath,
+    };
+    return new RollbarSourceMapPlugin(rollbarPluginOptions);
+};
+
+const validateRollbar = (environmentName, rollbarPlugin) => {
+    const requiresRollbar = environmentName === 'dogfood' || environmentName === 'staging';
+
+    if (!requiresRollbar) {
+        return;
+    }
+
+    if (!process.env[ROLLBAR_CLIENT_TOKEN_ENV_NAME]) {
+        throw new Error(`${ROLLBAR_CLIENT_TOKEN_ENV_NAME} must be set for ${environmentName}`);
+    }
+
+    if (!rollbarPlugin) {
+        if (environmentName === 'dogfood' || environmentName === 'staging') {
+            throw new Error(
+                `Please set rollbar env var ${ROLLBAR_PUBLISH_TOKEN_ENV_NAME} to a Rollbar project access token with post_server_item permissions to deploy source maps to ${environmentName}`,
+            );
```

Comment:
The final boolean check is redundant because we already know we are in a state of `requiresRollbar` I think.

Reply:
Ah good call, didn't remove this during a refactor.  Thanks for the sharp  👀